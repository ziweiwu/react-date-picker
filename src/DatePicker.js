import React from 'react';
import PropTypes from "prop-types";
import Icon from "@hig/icon";
import {TextFieldPresenter} from '@hig/text-field';
import '@hig/text-field/build/index.css';
import ReactDatePicker from 'react-datepicker';

export default class DatePicker extends React.Component {
  static propTypes = {
    adjustDateOnChange: PropTypes.bool,
    dateFormatCalendar: PropTypes.string,
    dayClassName: PropTypes.func,
    disabled: PropTypes.bool,
    endDate: PropTypes.object,
    focused: PropTypes.bool,
    highlightDates: PropTypes.array,
    id: PropTypes.string,
    locale: PropTypes.string,
    maxDate: PropTypes.object,
    minDate: PropTypes.object,
    onChange: PropTypes.func,
    selected: PropTypes.object,
    selectsEnd: PropTypes.bool,
    selectsStart: PropTypes.bool,
    startDate: PropTypes.object,
    startOpen: PropTypes.bool,
    title: PropTypes.string,
    todayButton: PropTypes.string,
    shouldCloseOnSelect: PropTypes.bool,
    readOnly: PropTypes.bool,
    showIcon: PropTypes.bool,
    instruction: PropTypes.string,
    showInstruction: PropTypes.bool,
    label: PropTypes.string,
    showLabel: PropTypes.bool,
    fixedHeight: PropTypes.bool,
    placeholder: PropTypes.string,
    showClearButton: PropTypes.bool,
    excludeDates: PropTypes.array,
    filterDate: PropTypes.func,
    dateFormat: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    includeDates: PropTypes.array,
    closeOnSelect: PropTypes.bool
  };

  static defaultProps = {
    focused: true,
    shouldCloseOnSelect: true,
    disabled: false,
    showIcon: true,
    label: undefined,
    showLabel: true,
    instruction: undefined,
    showInstruction: false,
    fixedHeight: true,
    showClearButton: false,
    today: true,
    closeOnSelect:true,
  };

  render() {
    const props = this.props;
    const calenderIcon = <Icon nameOrSVG="calendar"/>;
    const showIcon = this.props.showIcon ? calenderIcon : undefined;
    return (<ReactDatePicker
      {...props}
      className="hig__text-field-v1__input"
      id="hig__date-picker"
      ref={node => this.node = node}
      readOnly
      calender
      shouldCloseOnSelect={props.closeOnSelect}
      showMonthYearDropdown={false}
      showMonthDropdown={false}
      showYearDropdown={false}
      showTimeSelect={false}
      withPortal={false}
      isClearable={false}
      // use hig text field as inputField
      // use ref to allow the use clear button in text field component
      customInput={
        <TextFieldPresenter
          id={props.id}
          focused={props.focused}
          readOnly
          disabled
          placeholder={props.placeholder}
          instructions={props.showInstruction ? props.instruction : undefined}
          label={props.showLabel ? props.label : undefined}
          icon={showIcon}
          onFocus={null}
          onBlur={null}
          showClearButton={props.showClearButton}
          onClearButtonClick={() => {
            this.node.clear();
          }}
        />}

      // calender popper settings
      popperPlacement="bottom-start"
      popperModifiers={{
        // adjust position of calender popper, (horizontal, vertical)
        offset: {
          enabled: true,
          offset: '0px, 0px',
        },
        flip: {
          enabled: false,
        },
        preventOverflow: {
          enabled: true,
          escapeWithReference: false,
        },
      }}
    />);
  }
}

