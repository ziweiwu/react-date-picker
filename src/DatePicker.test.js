import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import DatePicker from './index';

describe('DatePicker', () => {
  // snapshot testing
  test('Snapshot tests regular view', () => {
    const wrapper = shallow(<DatePicker selected={moment('2017-09-15 09:30:00')} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot tests disabled view', () => {
    const wrapper = shallow(<DatePicker selected={moment('2017-09-15 09:30:00')} disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Snapshot tests calender Popper', () => {
    const wrapper = shallow(<DatePicker selected={moment('2017-09-15 09:30:00')} hidePopper={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  // test interactions
});
