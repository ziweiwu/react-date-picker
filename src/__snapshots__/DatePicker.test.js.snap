// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Snapshot tests calender Popper 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DatePicker
    allowSameDay={false}
    calender={true}
    className="hig__text-field-v1__input"
    closeOnSelect={true}
    customInput={
        <TextFieldPresenter
            disabled={true}
            focused={true}
            icon={
                <Icon
                    nameOrSVG="calendar"
                    size="24"
                />
            }
            id="text-field-36491"
            instructions={undefined}
            label={undefined}
            onBlur={null}
            onClearButtonClick={[Function]}
            onFocus={null}
            placeholder={undefined}
            readOnly={true}
            showClearButton={false}
            type="text"
        />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    fixedHeight={true}
    focused={true}
    hidePopper={false}
    id="hig__date-picker"
    isClearable={false}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
        Object {
            "flip": Object {
              "enabled": false,
            },
            "offset": Object {
              "enabled": true,
              "offset": "0px, 0px",
            },
            "preventOverflow": Object {
              "enabled": true,
              "escapeWithReference": false,
            },
          }
    }
    popperPlacement="bottom-start"
    preventOpenOnFocus={false}
    readOnly={true}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showClearButton={false}
    showIcon={true}
    showInstruction={false}
    showLabel={true}
    showMonthDropdown={false}
    showMonthYearDropdown={false}
    showTimeSelect={false}
    showYearDropdown={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
/>,
  "nodes": Array [
    <DatePicker
      allowSameDay={false}
      calender={true}
      className="hig__text-field-v1__input"
      closeOnSelect={true}
      customInput={
            <TextFieldPresenter
                  disabled={true}
                  focused={true}
                  icon={
                        <Icon
                              nameOrSVG="calendar"
                              size="24"
                        />
                  }
                  id="text-field-36491"
                  instructions={undefined}
                  label={undefined}
                  onBlur={null}
                  onClearButtonClick={[Function]}
                  onFocus={null}
                  placeholder={undefined}
                  readOnly={true}
                  showClearButton={false}
                  type="text"
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      focused={true}
      hidePopper={false}
      id="hig__date-picker"
      isClearable={false}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, 0px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      readOnly={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
      showMonthDropdown={false}
      showMonthYearDropdown={false}
      showTimeSelect={false}
      showYearDropdown={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DatePicker
      closeOnSelect={true}
      disabled={false}
      fixedHeight={true}
      focused={true}
      hidePopper={false}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
/>,
    "_forcedUpdate": false,
    "_instance": DatePicker {
      "context": Object {},
      "props": Object {
        "closeOnSelect": true,
        "disabled": false,
        "fixedHeight": true,
        "focused": true,
        "hidePopper": false,
        "selected": "2017-09-15T13:30:00.000Z",
        "shouldCloseOnSelect": true,
        "showClearButton": false,
        "showIcon": true,
        "showInstruction": false,
        "showLabel": true,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <DatePicker
      allowSameDay={false}
      calender={true}
      className="hig__text-field-v1__input"
      closeOnSelect={true}
      customInput={
            <TextFieldPresenter
                  disabled={true}
                  focused={true}
                  icon={
                        <Icon
                              nameOrSVG="calendar"
                              size="24"
                        />
                  }
                  id="text-field-36491"
                  instructions={undefined}
                  label={undefined}
                  onBlur={null}
                  onClearButtonClick={[Function]}
                  onFocus={null}
                  placeholder={undefined}
                  readOnly={true}
                  showClearButton={false}
                  type="text"
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      focused={true}
      hidePopper={false}
      id="hig__date-picker"
      isClearable={false}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, 0px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      readOnly={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
      showMonthDropdown={false}
      showMonthYearDropdown={false}
      showTimeSelect={false}
      showYearDropdown={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DatePicker
    closeOnSelect={true}
    disabled={false}
    fixedHeight={true}
    focused={true}
    hidePopper={false}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showClearButton={false}
    showIcon={true}
    showInstruction={false}
    showLabel={true}
/>,
}
`;

exports[`DatePicker Snapshot tests disabled view 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DatePicker
    allowSameDay={false}
    calender={true}
    className="hig__text-field-v1__input"
    closeOnSelect={true}
    customInput={
        <TextFieldPresenter
            disabled={true}
            focused={true}
            icon={
                <Icon
                    nameOrSVG="calendar"
                    size="24"
                />
            }
            id="text-field-36491"
            instructions={undefined}
            label={undefined}
            onBlur={null}
            onClearButtonClick={[Function]}
            onFocus={null}
            placeholder={undefined}
            readOnly={true}
            showClearButton={false}
            type="text"
        />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={true}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    fixedHeight={true}
    focused={true}
    id="hig__date-picker"
    isClearable={false}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
        Object {
            "flip": Object {
              "enabled": false,
            },
            "offset": Object {
              "enabled": true,
              "offset": "0px, 0px",
            },
            "preventOverflow": Object {
              "enabled": true,
              "escapeWithReference": false,
            },
          }
    }
    popperPlacement="bottom-start"
    preventOpenOnFocus={false}
    readOnly={true}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showClearButton={false}
    showIcon={true}
    showInstruction={false}
    showLabel={true}
    showMonthDropdown={false}
    showMonthYearDropdown={false}
    showTimeSelect={false}
    showYearDropdown={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
/>,
  "nodes": Array [
    <DatePicker
      allowSameDay={false}
      calender={true}
      className="hig__text-field-v1__input"
      closeOnSelect={true}
      customInput={
            <TextFieldPresenter
                  disabled={true}
                  focused={true}
                  icon={
                        <Icon
                              nameOrSVG="calendar"
                              size="24"
                        />
                  }
                  id="text-field-36491"
                  instructions={undefined}
                  label={undefined}
                  onBlur={null}
                  onClearButtonClick={[Function]}
                  onFocus={null}
                  placeholder={undefined}
                  readOnly={true}
                  showClearButton={false}
                  type="text"
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={true}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      focused={true}
      id="hig__date-picker"
      isClearable={false}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, 0px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      readOnly={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
      showMonthDropdown={false}
      showMonthYearDropdown={false}
      showTimeSelect={false}
      showYearDropdown={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DatePicker
      closeOnSelect={true}
      disabled={true}
      fixedHeight={true}
      focused={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
/>,
    "_forcedUpdate": false,
    "_instance": DatePicker {
      "context": Object {},
      "props": Object {
        "closeOnSelect": true,
        "disabled": true,
        "fixedHeight": true,
        "focused": true,
        "selected": "2017-09-15T13:30:00.000Z",
        "shouldCloseOnSelect": true,
        "showClearButton": false,
        "showIcon": true,
        "showInstruction": false,
        "showLabel": true,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <DatePicker
      allowSameDay={false}
      calender={true}
      className="hig__text-field-v1__input"
      closeOnSelect={true}
      customInput={
            <TextFieldPresenter
                  disabled={true}
                  focused={true}
                  icon={
                        <Icon
                              nameOrSVG="calendar"
                              size="24"
                        />
                  }
                  id="text-field-36491"
                  instructions={undefined}
                  label={undefined}
                  onBlur={null}
                  onClearButtonClick={[Function]}
                  onFocus={null}
                  placeholder={undefined}
                  readOnly={true}
                  showClearButton={false}
                  type="text"
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={true}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      focused={true}
      id="hig__date-picker"
      isClearable={false}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, 0px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      readOnly={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
      showMonthDropdown={false}
      showMonthYearDropdown={false}
      showTimeSelect={false}
      showYearDropdown={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DatePicker
    closeOnSelect={true}
    disabled={true}
    fixedHeight={true}
    focused={true}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showClearButton={false}
    showIcon={true}
    showInstruction={false}
    showLabel={true}
/>,
}
`;

exports[`DatePicker Snapshot tests regular view 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <DatePicker
    allowSameDay={false}
    calender={true}
    className="hig__text-field-v1__input"
    closeOnSelect={true}
    customInput={
        <TextFieldPresenter
            disabled={true}
            focused={true}
            icon={
                <Icon
                    nameOrSVG="calendar"
                    size="24"
                />
            }
            id="text-field-36491"
            instructions={undefined}
            label={undefined}
            onBlur={null}
            onClearButtonClick={[Function]}
            onFocus={null}
            placeholder={undefined}
            readOnly={true}
            showClearButton={false}
            type="text"
        />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    fixedHeight={true}
    focused={true}
    id="hig__date-picker"
    isClearable={false}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperModifiers={
        Object {
            "flip": Object {
              "enabled": false,
            },
            "offset": Object {
              "enabled": true,
              "offset": "0px, 0px",
            },
            "preventOverflow": Object {
              "enabled": true,
              "escapeWithReference": false,
            },
          }
    }
    popperPlacement="bottom-start"
    preventOpenOnFocus={false}
    readOnly={true}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showClearButton={false}
    showIcon={true}
    showInstruction={false}
    showLabel={true}
    showMonthDropdown={false}
    showMonthYearDropdown={false}
    showTimeSelect={false}
    showYearDropdown={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
/>,
  "nodes": Array [
    <DatePicker
      allowSameDay={false}
      calender={true}
      className="hig__text-field-v1__input"
      closeOnSelect={true}
      customInput={
            <TextFieldPresenter
                  disabled={true}
                  focused={true}
                  icon={
                        <Icon
                              nameOrSVG="calendar"
                              size="24"
                        />
                  }
                  id="text-field-36491"
                  instructions={undefined}
                  label={undefined}
                  onBlur={null}
                  onClearButtonClick={[Function]}
                  onFocus={null}
                  placeholder={undefined}
                  readOnly={true}
                  showClearButton={false}
                  type="text"
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      focused={true}
      id="hig__date-picker"
      isClearable={false}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, 0px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      readOnly={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
      showMonthDropdown={false}
      showMonthYearDropdown={false}
      showTimeSelect={false}
      showYearDropdown={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <DatePicker
      closeOnSelect={true}
      disabled={false}
      fixedHeight={true}
      focused={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
/>,
    "_forcedUpdate": false,
    "_instance": DatePicker {
      "context": Object {},
      "props": Object {
        "closeOnSelect": true,
        "disabled": false,
        "fixedHeight": true,
        "focused": true,
        "selected": "2017-09-15T13:30:00.000Z",
        "shouldCloseOnSelect": true,
        "showClearButton": false,
        "showIcon": true,
        "showInstruction": false,
        "showLabel": true,
      },
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <DatePicker
      allowSameDay={false}
      calender={true}
      className="hig__text-field-v1__input"
      closeOnSelect={true}
      customInput={
            <TextFieldPresenter
                  disabled={true}
                  focused={true}
                  icon={
                        <Icon
                              nameOrSVG="calendar"
                              size="24"
                        />
                  }
                  id="text-field-36491"
                  instructions={undefined}
                  label={undefined}
                  onBlur={null}
                  onClearButtonClick={[Function]}
                  onFocus={null}
                  placeholder={undefined}
                  readOnly={true}
                  showClearButton={false}
                  type="text"
            />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      fixedHeight={true}
      focused={true}
      id="hig__date-picker"
      isClearable={false}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperModifiers={
            Object {
                  "flip": Object {
                    "enabled": false,
                  },
                  "offset": Object {
                    "enabled": true,
                    "offset": "0px, 0px",
                  },
                  "preventOverflow": Object {
                    "enabled": true,
                    "escapeWithReference": false,
                  },
                }
      }
      popperPlacement="bottom-start"
      preventOpenOnFocus={false}
      readOnly={true}
      selected={"2017-09-15T13:30:00.000Z"}
      shouldCloseOnSelect={true}
      showClearButton={false}
      showIcon={true}
      showInstruction={false}
      showLabel={true}
      showMonthDropdown={false}
      showMonthYearDropdown={false}
      showTimeSelect={false}
      showYearDropdown={false}
      timeCaption="Time"
      timeIntervals={30}
      withPortal={false}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DatePicker
    closeOnSelect={true}
    disabled={false}
    fixedHeight={true}
    focused={true}
    selected={"2017-09-15T13:30:00.000Z"}
    shouldCloseOnSelect={true}
    showClearButton={false}
    showIcon={true}
    showInstruction={false}
    showLabel={true}
/>,
}
`;
